### DISCUSSION TITLE
Variable axes fonts

### DISCUSSION THREAD

**Discussion by dmalinovsky** [koreader/koreader#11688] [Q&A] [answered]
Does KOReader support the fonts with variable axes?  Take Google Fonts' Literata for example: https://fonts.google.com/specimen/Literata?query=literata

It supports 2 axes: weight and optical size. So instead of installing a font version with a specific weight, one can install a variable weight font and choose desired weight as needed.

I tried to find answer to this question online, but alas.

See here for more information: https://fonts.google.com/knowledge/introducing_type/introducing_variable_fonts

**Comment by dmalinovsky** [comment]
KOReader is using FreeType library, as I understand, and it supports OpenType Font Variations starting from [version 2.8](https://sourceforge.net/projects/freetype/files/freetype2/2.8/) (released in 2017).

P.S. Looks like it's version 2.13.2, so we should be good:
 
https://github.com/koreader/koreader-base/blob/865bed4c2f91c9808520fab24046685c409d8607/thirdparty/freetype2/CMakeLists.txt#L29-L30

**Reply by Frenzie** [reply]
I'm not sure if it's quite that simple, but yes, we use the latest FreeType. ^_^

**Reply by dmalinovsky** [reply]
Looks like [this comment](https://github.com/koreader/koreader/issues/4174#issuecomment-698520071) means that it's not so simple.

**Comment by poire-z** [comment, marked as answer]
Freetype may, but koreader and crengine don't.
It needs the user of FreeType to pilot the variation, so it means crengine would need to parse related CSS features, or detect font support, and adjust weight and italic... Lots of work for something that we don't really need.

I think I previously answered this question, but can't find it (may have been on these black holes that GitHub discussion or Gitter are).
